model/hoya_cell.hpp

```diff
@@ -213,6 +280,45 @@ class hoya_cell : public grid_cell<T, sir, mc> {
         }
         return new_r;
     }
+
+    std::vector<float> new_deaths(sir const &last_state) const {
+        std::vector<float> new_d = std::vector<float>();
+		float total_infected = 0;
+        for(int i = 0; i < n_age_segments(); i++) {
+			total_infected += last_state.infected[i];
+            new_d.push_back(last_state.infected[i] * mortality[i]);
+        }
+		if(total_infected > infected_capacity) {
+			for(int i = 0; i < n_age_segments(); i++) {
+				new_d[i] *= over_capacity_modifier;
```

Comment:
You must check that the number of deaths cannot be greater than the number of infected people (for every age segment)